helm/cluster-api/templates/delete-aggregated-roles/ciliumnetworkpolicy.yaml

```diff
@@ -2,26 +2,26 @@
 apiVersion: cilium.io/v2
 kind: CiliumNetworkPolicy
 metadata:
-  annotations:
-    helm.sh/hook: pre-install,pre-upgrade
-    helm.sh/hook-delete-policy: before-hook-creation,hook-succeeded
-    helm.sh/hook-weight: "-1" # before Job
+  name: delete-aggregated-roles
+  namespace: '{{ .Release.Namespace }}'
   labels:
-    application.giantswarm.io/branch: '{{ .Values.project.branch }}'
-    application.giantswarm.io/commit: '{{ .Values.project.commit }}'
-    app.kubernetes.io/instance: '{{ .Release.Name }}'
-    app.kubernetes.io/managed-by: '{{ .Release.Service }}'
+    helm.sh/chart: cluster-api
     app.kubernetes.io/name: cluster-api
+    app.kubernetes.io/instance: '{{ .Release.Name }}'
     app.kubernetes.io/version: '{{ .Chart.AppVersion }}'
-    helm.sh/chart: cluster-api
-  name: helm-hook-delete-aggregated-roles
-  namespace: '{{ .Release.Namespace }}'
+    app.kubernetes.io/managed-by: '{{ .Release.Service }}'
+    application.giantswarm.io/branch: '{{ .Values.project.branch }}'
+    application.giantswarm.io/commit: '{{ .Values.project.commit }}'
+  annotations:
+    helm.sh/hook: pre-install,pre-upgrade
+    helm.sh/hook-delete-policy: before-hook-creation,hook-succeeded
+    helm.sh/hook-weight: "-1"
 spec:
   endpointSelector:
     matchLabels:
       app.kubernetes.io/name: cluster-api
-      job: helm-hook-delete-aggregated-roles
```

Comment:
Is this right? It'll now match all pods created by this chart, not just the job?

Reply:
As this is a pre-install / pre-upgrade hook, it only applies for the time it's around. The NP gets removed after hook execution.

If we keep something like this, I'd prefer using `app.kubernetes.io/component`.

Reply:
Yeah, +1 for component label

Reply:
I added the component label everywhere now. I will try to extract this into helper templates in a second step.